packages/ember/src/components/form/select.gts

```diff
@@ -0,0 +1,412 @@
+      );
+    }
+
+    this.isOpen = true;
+  }
+
+  @action
+  onBlur() {
+    this.activeItem = -1;
+    this.isOpen = false;
+  }
+
+  get disabled() {
+    return this.args.disabled || this.args.loading;
+  }
+
+  @tracked
+  activeItem = -1;
+
+  @action
+  onKeyboardInput(evt: KeyboardEvent) {
+    if (!this.isOpen) {
+      return;
+    }
+    const key = evt.key.toLowerCase();
+    const isSingleChar = key.length === 1;
+    const isLetter = 'a' <= key && key <= 'z';
+    const isNumber = '0' <= key && key <= '9';
+    const isAlphaNumeric = (isLetter || isNumber) && isSingleChar;
+    if (!isAlphaNumeric) {
+      return;
+    }
+    evt.preventDefault();
+    evt.stopPropagation();
+    this.internalSearchBuffer += key;
+    const foundItem = this.selectItemBySearch();
+    if (!foundItem && this.internalSearchBuffer.length > 1) {
+      this.internalSearchBuffer = key;
+      this.selectItemBySearch();
+    }
+  }
+
+  @tracked
+  internalSearchBuffer = '';
+
+  selectItemBySearch() {
+    const searchBuffer = this.internalSearchBuffer;
+    if (searchBuffer.length === 0) {
+      return;
+    }
+
+    const childElements = Array.from(
+      document.querySelectorAll(`#${CSS.escape(this.menuId)} > li`),
+    );
+    for (const stringIndex in childElements) {
+      const index = parseInt(stringIndex);
+      const element = childElements[index];
+      if (!element) {
+        continue;
+      }
+      const text = element.textContent?.toLowerCase() ?? '';
+      const splitText = text.split(' ') ?? [];
+      for (const text of splitText) {
+        if (text.startsWith(searchBuffer)) {
+          this.activeItem = index;
+          this.scrollActiveItemIntoView();
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  scrollActiveItemIntoView() {
+    if (this.activeItem == -1) {
+      return;
+    }
+    const childElements = Array.from(
+      document.querySelectorAll(`#${CSS.escape(this.menuId)} > li`),
+    );
+    const activeElement = childElements[this.activeItem];
+    if (!activeElement) {
+      return;
+    }
+    activeElement.scrollIntoView({ block: 'nearest' });
+  }
+
+  @action
+  moveUp(evt: KeyboardEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+    this.internalSearchBuffer = '';
+    if (this.activeItem > 0) {
+      this.activeItem--;
+      this.scrollActiveItemIntoView();
+    }
+  }
+
+  @action
+  moveDown(evt: KeyboardEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+    this.internalSearchBuffer = '';
+    if (this.activeItem < this.internalOptions.length - 1) {
+      this.activeItem++;
+      this.scrollActiveItemIntoView();
+    }
+  }
+
+  @action
+  enterKeyHandler(evt?: KeyboardEvent) {
+    evt?.preventDefault();
+    evt?.stopPropagation();
+    this.internalSearchBuffer = '';
+    if (!this.isOpen) {
+      this.onFocus();
+    } else {
+      this.exitKeyHandler();
+    }
+  }
+
+  @action
+  exitKeyHandler(evt?: KeyboardEvent) {
```

Comment:
Should the exit select the currently highlighted option, or should it just close the dropdown? According to ChatGPT (so take it with a hint of skepticism):
> When using the browser-native `select` element, if an option is highlighted and the Escape key is pressed, the `select` element closes without selecting the highlighted option. The original selection (if any) remains unchanged. The Escape key acts as a cancel action, effectively dismissing the dropdown menu without making any changes to the selection.

Reply:
The current nrg dropdown selects the highlighted item on `escape`.
However, the robot makes a valid point.
Should we change this behavior moving forward?

Reply:
While this is a different behavior than the existing `ember-nrg-ui`, I do think treating `Escape` as a "cancel" action makes more sense.